src/rate-limiting/fixed.ts

```diff
@@ -0,0 +1,64 @@
+import { AdapterRateLimitTier, RateLimiter } from '.'
+import { AdapterEndpoint, EndpointGenerics } from '../adapter'
+import { makeLogger, sleep } from '../util'
+
+const logger = makeLogger('FixedRateLimiter')
+
+/**
+ * The simplest version of a rate limit. This will not take any bursts into accoung,
+ * and always rely on a fixed request per second rate. The only "complex" mechanism it has
+ * is checking when the last request was made to this rate limiter, to account for a period
+ * of time with no requests and avoiding the wait of the initial request.
+ */
+export class FixedRateLimiter implements RateLimiter {
+  period!: number
+  lastRequestAt: number | null = null
+
+  initialize<T extends EndpointGenerics>(
+    endpoints: AdapterEndpoint<T>[],
```

Comment:
Do we still need endpoints in here? Looks like neither rate limiter uses it during initialization

Reply:
Not in the current state, but there's code to allow for endpoint allocation that we might want to use in the future, if we want to clean it up I'd say we do that later

Reply:
Fair enough